### ISSUE TITLE
🌟

### ISSUE THREAD

**Issue by justusweigmann** [HiEventsDev/Hi.Events#186]
**Is your feature request related to a problem? Please describe.**
Hi.Events currently lacks a solution for customers who cannot print their own tickets or don't have smartphones to display digital tickets. This limits accessibility and potentially reduces ticket sales. Additionally, we're missing an opportunity for an upsell service that could benefit both our business and our customers.

**Describe the solution you'd like**
We propose adding a feature to the admin interface that allows for printing QR codes of tickets, potentially including booking numbers and ticket tiers, sized to fit Avery Zweckform 30mm x 20mm labels. This feature would enable:

1. Selecting specific tickets for QR code printing
2. Choosing to include additional information (booking number, ticket tier)
3. Generating printable QR codes formatted for the specified label size
4. Batch printing of multiple QR codes

**Describe alternatives you've considered**
1. Offering only digital tickets and not catering to customers without smartphones
2. Printing each qr code on it's own (labour intense task)

**Additional context**
This feature would create a new revenue stream through an upsell service while improving accessibility for customers without printing capabilities or smartphones. It would be particularly useful for events with older demographics or in areas with limited technology access.

**Comment by daveearley** [comment]
Badge printing is definitely something we want to support. I've considered leveraging [ZPL](https://developer.zebra.com/products/printers/zpl), which is widely used for printing on Zebra printers. However, since ZPL is specific to Zebra hardware, it limits compatibility with other printers. Using PDFs seems like a more logical option.

I'd love to eventually have WYSIWYG badge designer, but that would be a large engineering effort.

**Comment by justusweigmann** [comment]
Hi @daveearley,

I'm glad to hear this is on your roadmap.

From what I understand, you can print the PDF directly to the Zebra printer. If you're looking to implement an on-demand printing service that connects directly to the printer, using CUPS might be the best option.

Having a WYSIWYG feature would be fantastic. However, to start, a simple label design like the one we currently use would suffice. I might have some time to help get this started, as we're eager to move away from our old, unreliable system.

<img width="1023" alt="Screenshot 2024-09-04 at 09 58 42" src="https://github.com/user-attachments/assets/69b9bb9f-6b88-42dc-830c-1587a26f1356">

**Comment by amican** [comment]
Can't wait for this feature, maybe for badge printing too
